conf/modules.config

```diff
@@ -1,337 +1,836 @@
+            ]
         }
+    }
+}
 
-        "ucsc_bedclip" {
-            suffix        = ".clipped"
-            publish_dir   = "03_peak_calling/02_clip_bed"
+if(params.run_trim_galore_fastqc && !params.skip_trimming) {
+    process {
+        withName: '.*:FASTQC_TRIMGALORE:TRIMGALORE' {
+            ext.args   = [
+                '--fastqc',
+                params.trim_nextseq > 0 ? "--nextseq ${params.trim_nextseq}" : ''
+            ].join(' ').trim()
+            publishDir = [
+                [
+                    path: { "${params.outdir}/01_prealign/trimgalore/fastqc" },
+                    mode: 'copy',
+                    pattern: "*.{html,zip}"
+                ],
+                [
+                    path: { "${params.outdir}/01_prealign/trimgalore" },
+                    mode: 'copy',
+                    pattern: "*.fastq.gz",
+                    enabled: params.save_trimmed
+                ],
+                [
+                    path: { "${params.outdir}/01_prealign/trimgalore" },
+                    mode: 'copy',
+                    pattern: "*.txt"
+                ]
+            ]
         }
+    }
+}
+
+/*
+========================================================================================
+    ALIGNMENT
+========================================================================================
+*/
 
-        "ucsc_bedgraphtobigwig" {
-            publish_dir   = "03_peak_calling/03_bed_to_bigwig"
+if(params.run_alignment) {
+    process {
+        withName: '.*:ALIGN_BOWTIE2:BOWTIE2_ALIGN' {
+            ext.args   = { params.skip_trimming ? '--end-to-end --very-sensitive --no-mixed --no-discordant --phred33 -I 10 -X 700' : '--local --very-sensitive --no-mixed --no-discordant --phred33 -I 10 -X 700' }
+            publishDir = [
+                [
+                    path: { "${params.outdir}/02_alignment/${params.aligner}/target/log" },
+                    mode: 'copy',
+                    pattern: '*.log'
+                ],
+                [
+                    path: { "${params.outdir}/02_alignment/${params.aligner}/target" },
+                    mode: 'copy',
+                    pattern: '*.bam',
+                    enabled: ( params.save_align_intermed || (!params.run_q_filter && !params.run_mark_dups && !params.run_remove_dups) )
+                ],
+                [
+                    path: { "${params.outdir}/02_alignment/${params.aligner}/target/unmapped" },
+                    mode: 'copy',
+                    pattern: '*.fastq.gz',
+                    enabled: params.save_unaligned
+                ]
+            ]
+        }
+
+        withName: '.*:ALIGN_BOWTIE2:BOWTIE2_SPIKEIN_ALIGN' {
+            ext.args   = { params.skip_trimming ? '--end-to-end --very-sensitive --no-overlap --no-dovetail --no-mixed --no-discordant --phred33 -I 10 -X 700' : '--local --very-sensitive --no-overlap --no-dovetail --no-mixed --no-discordant --phred33 -I 10 -X 700' }
```

Comment:
could it be useful to put the `bowtie2` parameters in the params scope so that the user can update them in-the-fly if necessary ?

Reply:
These can be customised already using an extra config file. To break out bowtie2 params into separate config params would be too difficult i think